javaharness/java/arcs/android/impl/AndroidHarnessController.java

```diff
@@ -52,15 +60,21 @@ public WebResourceResponse shouldInterceptRequest(WebView view, WebResourceReque
         return super.shouldInterceptRequest(view, request);
       }
     });
-    webView.loadUrl("file:///android_asset/index.html?log");
 
-    // TODO: add a boolean flag to controll these.
+    RuntimeSettings settings = runtimeSettings.get();
 
-    // Uncomment this to view Arcs in devtools extension:
-    // webView.loadUrl("file:///android_asset/index.html?explore-proxy");
-    // Uncomment this to load pipes-shell from localhost and to view Arcs in devtools extension:
-    // webView.loadUrl("http://localhost:8786/shells/pipes-shell/web/deploy/dist/?log=2&explore-proxy");
-    // Also add to <application> in service/AndroidManifest.xml:
+    // If using the host pipes-shell, adding the following to allow HTTP connection(s) at
+    // <application> in service/AndroidManifest.xml:
     //    android:usesCleartextTraffic="true"
+    String url = settings.useHostPipesShell()
+        ? "http://localhost:8786/shells/pipes-shell/web/deploy/dist/?"
```

Comment:
optional - can this be a string instead of a boolean? this way the entire url can be provided via the settings and not hardcoded...

Reply:
yes yes, good suggestion, i will change it to a string instead since it gives us the flexibility of adopting any shell in the future.
The one thing I'd like to head up is that there is a limitation on the length of android property, it's like 256 chars or else I can't remember.

Reply:
Done.